Added hotkeys for support powers

sinf opened the pull request:
Implemented hotkey logic for support powers (as requested in #7369) and UI items for configuring the hotkeys.
Also the hotkey is displayed in the support power tooltip in the same manner as unit/structure tooltips are.

Phrohdoh commented:
```
OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs:L124: [SA1513] Statements or elements wrapped in curly brackets must be followed by a blank line.
OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs:L121: [SA1008] Invalid spacing around the opening parenthesis.
OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs:L121: [SA1009] Invalid spacing around the closing parenthesis.
OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs:L65: [SA1507] The code must not contain multiple blank lines in a row.
```

Phrohdoh commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs:
```diff
@@ -64,4 +76,4 @@ public SupportPowerTooltipLogic(Widget widget, TooltipContainerWidget tooltipCon
 	}
-}
\ No newline at end of file
+}
```
This hunk will need to be reverted.

Mailaender commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/SupportPowerTooltipLogic.cs:
```diff
@@ -35,10 +36,13 @@ public SupportPowerTooltipLogic(Widget widget, TooltipContainerWidget tooltipCon
 			{
-				var sp = palette.TooltipPower;
-				if (sp == null)
+				var spicon = palette.TooltipIcon;
```
The naming confused me. Maybe just call it `icon`?

sinf committed:
Added Hotkeys for support powers

abcdefg30 commented:
Works ingame and code looks ok to me. :+1: / :white_check_mark:

pchote commented on mods/cnc/chrome/settings.yaml:
```diff
@@ -451,6 +451,20 @@ Container@SETTINGS_PANEL:
 											X: PARENT_RIGHT-WIDTH+1
 											Width: 80
 											Height: 25
+								Container@SUPPORTPOWER_TEMPLATE:
```
These are just duplicating the production template.  Could you reuse that instead?

pchote commented:
TD uses a smaller margin than RA, so offsets can't be copied directly between the two mods:

![support](https://cloud.githubusercontent.com/assets/167819/7331959/0fdcab24-eb7c-11e4-9d88-bdc259ffaa22.png)

pchote commented:
If we use shift+F1-6 for support powers then we will not be able to fix #5544.  Are people happy with this tradeoff?

sinf commented:
Reused the production template thingy. Removed X coordinate from the cnc tooltip so its like the other tooltips there. Also changed the default bindings to "Unknown" so they don't conflict with anything. If a user doesn't want a feature then it shouldn't be in the way. But if a user does want to use the feature then it can easily be enabled. Isn't this the best for everyone?

Unit158 commented:
Why not just shift+number?

On 2015-04-25, at 2:52 AM, Paul Chote <notifications@github.com> wrote:

> If we use shift+F1-6 for support powers then we will not be able to fix #5544. Are people happy with this tradeoff?
> 
> ―
> Reply to this email directly or view it on GitHub.
>

penev92 commented:
Because that is already used for something else.

Unit158 commented:
Are you sure? I didn't see it bound to anything.

On 2015-04-25, at 7:12 PM, Pavel Penev <notifications@github.com> wrote:

> Because that is already used for something else.
> 
> ―
> Reply to this email directly or view it on GitHub.
>

penev92 commented:
Numbers are used for control groups. Shift is used to modify control groups. Shift + X means "add control group X to the current selection".

Unit158 commented:
That's ctrl+#. I'm pretty sure they're unused.

On 2015-04-26, at 6:06 AM, Pavel Penev <notifications@github.com> wrote:

> Numbers are used for control groups. Shift is used to modify control groups. Shift + X means "add control group X to the current selection".
> 
> ―
> Reply to this email directly or view it on GitHub.
>

penev92 commented:
@pchote: I'm perfectly fine with not having that. Support power hotkeys are immensely more important.
@Unit158: I'm not going to argue over this. Go play <insert_any_RTS_game's_name_here> and see for yourself.

abcdefg30 commented:
> @pchote: I'm perfectly fine with not having that. Support power hotkeys are immensely more important.

Sorry, but I'm not.

Unit158 commented:
Shift+# is unused. Feel free to use it.

penev92 commented:
Seriously, I don't feel like explaining in both IRC and here. Can we drop the spam from here?
Shift + number is one of the basic key combinations for unit management in any RTS game ever, and we have it.

Unit158 commented:
On a US English keyboard, Shift+# does jack shit. I'll even upload a video. It is ctrl+# that creates and adds units to control groups.

abcdefg30 commented:
Let's stay with what @sinf suggested (unkown keycode).

Mailaender commented:
:+1: Works as promised. Thanks!

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/acf34bd1c87e7cc52f99778f83413c4c89439ed9...7e447a50da156681ac3b4910a5fcccd546978fc4)
